frontend/catalyst/jax_primitives.py

```diff
@@ -937,34 +945,62 @@ def _gphase_def_impl(*args, **kwargs):
 
 
 def _gphase_lowering(
-    jax_ctx: mlir.LoweringRuleContext, *qubits_or_params, ctrl_len=0, adjoint=False
+    jax_ctx: mlir.LoweringRuleContext,
+    *qubits_or_params,
+    ctrl_len=0,
+    adjoint=False,
+    static_params=None,
 ):
     ctx = jax_ctx.module_context.context
     ctx.allow_unregistered_dialects = True
 
-    param = qubits_or_params[0]
-    ctrl_qubits = qubits_or_params[1 : 1 + ctrl_len]
-    ctrl_values = qubits_or_params[1 + ctrl_len :]
+    param = None if static_params else qubits_or_params[-1]
+    ctrl_qubits = qubits_or_params[:ctrl_len]
+    ctrl_values = qubits_or_params[ctrl_len:-1] if param else qubits_or_params[ctrl_len:]
+
+    assert (
+        not static_params or len(static_params) == 1
+    ), "GlobalPhase only takes one static float parameter"
+    param_attr = (
```

Comment:
It's easy to miss the assignment here:
```suggestion
    ), "GlobalPhase only takes one static float parameter"

    param_attr = (
```